gammapy/maps/region.py

```diff
@@ -291,6 +295,35 @@ def get_wcs_coord(self):
         region_coord = wcs_geom.get_coord().apply_mask(common_coord)
         return region_coord
 
+    def get_wcs_weights(self):
+        """Get an array of weights that represent the
+            fraction of each pixel from the minimal
+            equivalent geometry contained in the region.
+
+        Returns
+        -------
+        weights : `~Map`
+        """
+        factor = 10
+        wcs_geom = self.to_wcs_geom()
+
+        # if the region has non-spatial axis, this way is faster
+        # create the spatial weights only
+        wcs_geom_upsampled = wcs_geom.to_image().upsample(factor=factor)
+        data = self.contains(wcs_geom_upsampled.get_coord()).astype(float)
+        weights_no_axis = Map.from_geom(wcs_geom_upsampled, data=data)
+        weights_no_axis = weights_no_axis.downsample(factor=factor)
+        weights_no_axis.data /= weights_no_axis.data.max()
+
+        # And then make the map with the right dimensions
+        data = weights_no_axis.data
+        for axis in self.axes:
+            data = np.stack(axis.nbin*[data])
+
+        weights = Map.from_geom(wcs_geom, data=data)
```

Comment:
I think typically there is no need to store the weights in a map, I did in my example just to visualise it...

Reply:
good point :+1: